client/ayon_traypublisher/plugins/create/create_csv_ingest.py

```diff
@@ -835,6 +895,23 @@ def _create_instances_from_csv_data(self, csv_dir: str, filename: str):
                 "prepared_data_for_repres": []
             }
 
+            if product_item.has_promised_context:
+                hierarchy, _ = folder_path.rsplit("/",1)
+                families.append("csv_ingest_shot")
+                instance_data.update(
+                    {
+                        "newHierarchyIntegration": True,
+                        "hierarchy": hierarchy,
+                        "parents": product_item.parents
+                    }
+                )
+                # TODO create new task from provided task name
+#                if product_item.task_name:
+#                    instance_data["tasks"] = {
+#                        "name": product_item.task_name,
+#                        "type": "Generic"
+#                    }
```

Comment:
```suggestion
                if product_item.task_name:
                    instance_data["tasks"] = [
	                    {
	                        "name": product_item.task_name,
	                        "type": "Generic"
	                    }
                    ]
```

Reply:
```suggestion
                if product_item.task_name:
                    tasks = instance_data.setdefault("tasks", [])
                    tasks.append(
                        {
                            "name": product_item.task_name,
                            "type": "Generic"
                        }
                    )
```

Or since I am not sure if tasks could be added in batch we could perhaps do it this way.